column_index.go

```diff
@@ -676,19 +676,29 @@ func truncateLargeMinByteArrayValue(value []byte, sizeLimit int) []byte {
 }
 
 func truncateLargeMaxByteArrayValue(value []byte, sizeLimit int) []byte {
-	if len(value) > sizeLimit && !isMaxByteArrayValue(value) {
-		value = value[:sizeLimit]
+	if len(value) > sizeLimit {
+		if newValue := incrementByteArray(value[:sizeLimit]); newValue != nil {
+			value = newValue
+		}
```

Comment:
This changes now that we don't allocate.
```suggestion
		value = value[:sizeLimit]
		incrementByteArray(value)
```